xayn-ai/src/ltr/mod.rs

```diff
@@ -116,6 +119,86 @@ impl LtrSystem for ConstLtr {
     }
 }
 
+pub type OwnedSample = (Array2<f32>, Array1<f32>);
+
+/// Creates training data for ListNet from a users history.
+pub fn list_net_training_data_from_history(
+    mut history: &[DocumentHistory],
+) -> Result<Vec<OwnedSample>, Error> {
```

Comment:
This implementation is far from perfect, but will be replaced anyway in a follow-up PR.